packages/edit-post/src/components/header/template-title/delete-template.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { pickBy } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { MenuGroup, MenuItem } from '@wordpress/components';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { store as editPostStore } from '../../../store';
+
+export default function DeleteTemplate() {
+	const { clearSelectedBlock } = useDispatch( blockEditorStore );
+	const { setIsEditingTemplate } = useDispatch( editPostStore );
+	const { getEditorSettings } = useSelect( editorStore );
+	const { updateEditorSettings, editPost } = useDispatch( editorStore );
+	const { deleteEntityRecord } = useDispatch( coreStore );
+	const { template } = useSelect( ( select ) => {
+		const { isEditingTemplate, getEditedPostTemplate } = select(
+			editPostStore
+		);
+		const _isEditing = isEditingTemplate();
+		return {
+			template: _isEditing ? getEditedPostTemplate() : null,
+		};
+	}, [] );
+
+	if ( ! template || ! template.wp_id ) {
+		return null;
+	}
+	return (
+		<MenuGroup className="edit-post-template-top-area__second-menu-group">
+			<MenuItem
+				isDestructive
+				isTertiary
+				isLink
+				onClick={ () => {
+					if (
+						// eslint-disable-next-line no-alert
+						window.confirm(
+							__(
+								'Are you sure you want to delete this template? It may be currently in use by other pages or posts.'
+							)
```

Comment:
```suggestion
							/* translators: %1$s: template name */
							sprintf(
								'Are you sure you want to delete the %s template? It may be in use by other pages or posts.',
								templateTitle
							)
```
Add the template name to the confirmation message. Simplify the sentence.
-This order of `template name` followed by the word `template` may be difficult to translate correctly. 🤔

Reply:
"It may be in use by" doesn't read quite right to me. "It may be used by" sounds a bit better IMO.